src/Umbraco.Forms.Integrations.Crm.ActiveCampaign/ActiveCampaignContactsWorkflow.cs

```diff
@@ -0,0 +1,137 @@
+﻿using Microsoft.Extensions.Options;
+
+using System.Text.Json;
+
+using Umbraco.Forms.Core;
+using Umbraco.Forms.Core.Enums;
+using Umbraco.Forms.Core.Persistence.Dtos;
+using Umbraco.Forms.Integrations.Crm.ActiveCampaign.Configuration;
+using Umbraco.Forms.Integrations.Crm.ActiveCampaign.Models.Dtos;
+using Umbraco.Forms.Integrations.Crm.ActiveCampaign.Services;
+
+namespace Umbraco.Forms.Integrations.Crm.ActiveCampaign
+{
+    public class ActiveCampaignContactsWorkflow : WorkflowType
+    {
+        private readonly ActiveCampaignSettings _settings;
+
+        private readonly IAccountService _accountService;
+
+        private readonly IContactService _contactService;
+
+        [Core.Attributes.Setting("Account",
+            Description = "Please select an account",
+            View = "~/App_Plugins/UmbracoForms.Integrations/Crm/ActiveCampaign/accountpicker.html")]
+        public string Account { get; set; }
+
+        [Core.Attributes.Setting("Contact Mappings",
+            Description = "Map contact details with form fields",
+            View = "~/App_Plugins/UmbracoForms.Integrations/Crm/ActiveCampaign/contact-mapper.html")]
+        public string ContactMappings { get; set; }
+
+        [Core.Attributes.Setting("Custom Field Mappings",
+            Description = "Map contact custom fields with form fields",
+            View = "~/App_Plugins/UmbracoForms.Integrations/Crm/ActiveCampaign/customfield-mapper.html")]
+        public string CustomFieldMappings { get; set; }
+
+        public ActiveCampaignContactsWorkflow(IOptions<ActiveCampaignSettings> options, IAccountService accountService, IContactService contactService)
+        {
+            Id = new Guid(Constants.WorkflowId);
+            Name = "ActiveCampaign Contacts Workflow";
+            Description = "Submit form data to ActiveCampaign Contacts";
+            Icon = "icon-users";
+
+            _settings = options.Value;
+
+            _accountService = accountService;
+
+            _contactService = contactService;
+        }
+
+        public override WorkflowExecutionStatus Execute(WorkflowExecutionContext context)
+        {
+            var mappings = JsonSerializer.Deserialize<List<ContactMappingDto>>(ContactMappings);
+
+            var email = context.Record.RecordFields[Guid.Parse(mappings.First(p => p.ContactField == "email").FormField.Id)]
+                .ValuesAsString();
+
+            // Check if contact exists.
+            var contacts = _contactService.Get(email).Result;
```

Comment:
Careful here in case we get a deadlock issue as we had before with converting an async request to a sync one.  You might want to add `ConfigureAwait()` and use `GetAwaiter().GetResult().